lms/djangoapps/instructor/views/api.py

```diff
@@ -1506,23 +1506,36 @@ def reset_student_attempts(request, course_id):
         if not has_access(request.user, 'instructor', course):
             return HttpResponseForbidden("Requires instructor access.")
 
-    try:
-        module_state_key = course_id.make_usage_key_from_deprecated_string(problem_to_reset)
-    except InvalidKeyError:
-        return HttpResponseBadRequest()
+    if problem_to_reset != 'entrance_exam':
+        try:
+            module_state_key = course_id.make_usage_key_from_deprecated_string(problem_to_reset)
+        except InvalidKeyError:
+            return HttpResponseBadRequest()
 
     response_payload = {}
     response_payload['problem_to_reset'] = problem_to_reset
 
     if student:
-        try:
-            enrollment.reset_student_attempts(course_id, student, module_state_key, delete_module=delete_module)
-        except StudentModule.DoesNotExist:
-            return HttpResponseBadRequest(_("Module does not exist."))
-        except sub_api.SubmissionError:
-            # Trust the submissions API to log the error
-            error_msg = _("An error occurred while deleting the score.")
-            return HttpResponse(error_msg, status=500)
+        if problem_to_reset == 'entrance_exam':
+            entrance_exam_modules = get_entrance_exam_modules(request, course, student)
```

Comment:
I believe get_entrance_exam_modules can raise an ItemNotFoundError exception, if the entrance exam key is pointing to something that cannot be resolved. It seems like we should have some exception handling here and a HttpResponseBadRequest() returned to the Ajax client to display an error message.